### PR TITLE
Update psse parser for 3winding transformers

### PR DESCRIPTION
**Pull request by hei06j**
This PR is to address the issue #912, which also partly addresses the issue #888. The issue #888 still stands, but with this PR, it solves by changing the transformer sbase to >70 (it is 60 now and not solving).

PSSE three-winding transformer supports 
- zero NOMVx values, which in this case, the transformer x-th winding nominal voltage is the same as the bus basekV it is connected to,
- cw = 1 (for off-nominal turns ratio in pu of winding bus base voltage)
- cw = 2 (for winding voltage in kv)
- cw = 3 (for off-nominal turns ratio in pu of nominal winding voltage, NOMV1, NOMV2 and NOMV3.)

which are now supported in the PSSE parser as part of this PR.

### CODE REVIEW

File: src/io/psse.jl [hunk 1]
```diff
@@ -601,9 +632,12 @@ function _psse2pm_transformer!(pm_data::Dict, pti_data::Dict, import_all::Bool)
...
-                        if transformer["CW"] == 3  # "for off-nominal turns ratio in pu of nominal winding voltage, NOMV1, NOMV2 and NOMV3."
-                            sub_data["tap"] *= transformer["NOMV$m"]
+                        if transformer["CW"] == 2
+                            sub_data["tap"] /= _get_bus_value(bus_id, "base_kv", pm_data)
+                        elseif transformer["CW"] == 3  # "for off-nominal turns ratio in pu of nominal winding voltage, NOMV1, NOMV2 and NOMV3."
+                            if !iszero(transformer["NOMV$m"])
+                                sub_data["tap"] *= transformer["NOMV$m"] / _get_bus_value(bus_id, "base_kv", pm_data)
+                            end
                         end
                     end
```
Comment (ccoffrin) [inline src/io/psse.jl:642]: Could you elaborate about what the correct semantics of this should be?

The way the code is organized seems risky to me.  For example, what happens if `transformer["CW"] == 3` and `iszero(transformer["NOMV$m"]) == true`

Comment (hei06j) [reply src/io/psse.jl:642]: `iszero(transformer["NOMV$m"]) == true` means that the transformer winding `m` nominal voltage is the same as the bus base voltage it is connected to, i.e. `_get_bus_value(bus_id, "base_kv", pm_data)`. So in the case of `transformer["CW"] == 3`, the tap value won't change. For when `transformer["CW"] == 2`, the winding voltage is in kv and does not need to account for `transformer["NOMV$m"]`.

Comment (ccoffrin) [reply src/io/psse.jl:642]: Given the complexity of this sepecific section I suggest we re-write the code int he form,

```
if transformer["CW"] == 1
  ...
elseif transformer["CW"] == 2
  ...
elseif transformer["CW"] == 3
  ...
else
  error - unsupported value for "CW" on transformer ....
end
```

Comment (hei06j) [reply src/io/psse.jl]: Recommendation implemented. :)

File: src/io/psse.jl [hunk 2]
```diff
@@ -544,13 +562,29 @@ function _psse2pm_transformer!(pm_data::Dict, pti_data::Dict, import_all::Bool)
 
                 # Unit Transformations
                 if transformer["CZ"] != 1  # NOT "for resistance and reactance in pu on system MVA base and winding voltage base"
-                    br_r12 *= (transformer["NOMV1"] / _get_bus_value(bus_id1, "base_kv", pm_data))^2 * (pm_data["baseMVA"] / transformer["SBASE1-2"])
-                    br_r23 *= (transformer["NOMV2"] / _get_bus_value(bus_id2, "base_kv", pm_data))^2 * (pm_data["baseMVA"] / transformer["SBASE2-3"])
-                    br_r31 *= (transformer["NOMV3"] / _get_bus_value(bus_id3, "base_kv", pm_data))^2 * (pm_data["baseMVA"] / transformer["SBASE3-1"])
+                    if transformer["NOMV1"] == 0.0
```
Comment (ccoffrin) [inline src/io/psse.jl:565]: same note as above.

File: src/io/psse.jl [hunk 3]
```diff
@@ -544,13 +562,29 @@ function _psse2pm_transformer!(pm_data::Dict, pti_data::Dict, import_all::Bool)
...
+                        br_x12 *= (pm_data["baseMVA"] / transformer["SBASE1-2"])
+                    else
+                        br_r12 *= (transformer["NOMV1"] / _get_bus_value(bus_id1, "base_kv", pm_data))^2 * (pm_data["baseMVA"] / transformer["SBASE1-2"])
+                        br_x12 *= (transformer["NOMV1"] / _get_bus_value(bus_id1, "base_kv", pm_data))^2 * (pm_data["baseMVA"] / transformer["SBASE1-2"])
+                    end
 
-                    br_x12 *= (transformer["NOMV1"] / _get_bus_value(bus_id1, "base_kv", pm_data))^2 * (pm_data["baseMVA"] / transformer["SBASE1-2"])
-                    br_x23 *= (transformer["NOMV2"] / _get_bus_value(bus_id2, "base_kv", pm_data))^2 * (pm_data["baseMVA"] / transformer["SBASE2-3"])
-                    br_x31 *= (transformer["NOMV3"] / _get_bus_value(bus_id3, "base_kv", pm_data))^2 * (pm_data["baseMVA"] / transformer["SBASE3-1"])
+                    if transformer["NOMV2"] == 0.0
```
Comment (ccoffrin) [inline src/io/psse.jl:573]: same note as above.

File: src/io/psse.jl [hunk 4]
```diff
@@ -544,13 +562,29 @@ function _psse2pm_transformer!(pm_data::Dict, pti_data::Dict, import_all::Bool)
...
+                        br_r23 *= (pm_data["baseMVA"] / transformer["SBASE2-3"])
+                        br_x23 *= (pm_data["baseMVA"] / transformer["SBASE2-3"])
+                    else
+                        br_r23 *= (transformer["NOMV2"] / _get_bus_value(bus_id2, "base_kv", pm_data))^2 * (pm_data["baseMVA"] / transformer["SBASE2-3"])
+                        br_x23 *= (transformer["NOMV2"] / _get_bus_value(bus_id2, "base_kv", pm_data))^2 * (pm_data["baseMVA"] / transformer["SBASE2-3"])
+                    end
+                    
+                    if transformer["NOMV3"] == 0.0
```
Comment (ccoffrin) [inline src/io/psse.jl:581]: same note as above

**Review by ccoffrin — COMMENTED** [review]
Ok, I think this is very close to ready. I do have one last question about float equality testing to resolve.

**Comment by ccoffrin** [comment]
Looks good.  Thanks for the contribution!

**Pull request merged**